packages/dds/matrix/src/test/memory/matrix.spec.ts

```diff
@@ -1,155 +0,0 @@
-/*!
- * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
- * Licensed under the MIT License.
- */
-
-import {
-	type IMemoryTestObject,
-	benchmarkMemory,
-	isInPerformanceTestingMode,
-} from "@fluid-tools/benchmark";
-import type { IChannel } from "@fluidframework/datastore-definitions/internal";
-import { MockFluidDataStoreRuntime } from "@fluidframework/test-runtime-utils/internal";
-
-import { SharedMatrix, type ISharedMatrix } from "../../index.js";
-import { matrixFactory } from "../utils.js";
-
-function createLocalMatrix(id: string): ISharedMatrix & IChannel {
-	return matrixFactory.create(new MockFluidDataStoreRuntime(), id);
-}
-
-describe("Matrix memory usage", () => {
-	// IMPORTANT: variables scoped to the test suite are a big problem for memory-profiling tests
-	// because they won't be out of scope when we garbage-collect between runs of the same test,
-	// and that will skew measurements. Tests should allocate all the memory they need using local
-	// variables scoped to the test function itself, so several iterations of a given test can
-	// measure from the same baseline (as much as possible).
-
-	beforeEach(async () => {
-		// CAREFUL: usually beforeEach/afterEach hooks are used to initialize or interact with variables
-		// whose scope is the encompasing test suite, but that's a problem for memory-profiling tests.
-		// See the comment at the top of the test suite for more details.
-	});
-
-	afterEach(() => {
-		// CAREFUL: usually beforeEach/afterEach hooks are used to initialize or interact with variables
-		// whose scope is the encompasing test suite, but that's a problem for memory-profiling tests.
-		// See the comment at the top of the test suite for more details.
-	});
-
-	describe("Detached Matrix", () => {
-		benchmarkMemory(
-			new (class implements IMemoryTestObject {
-				readonly title = "Create empty Matrix";
-				minSampleCount = 500;
-
-				private localMatrix: SharedMatrix = createLocalMatrix("testLocalMatrix");
-
-				async run(): Promise<void> {
-					this.localMatrix = createLocalMatrix("testLocalMatrix");
-				}
-			})(),
-		);
-
-		const numbersOfEntriesForTests = isInPerformanceTestingMode
-			? [100, 1000, 5000]
-			: // When not measuring perf, use a single smaller data size so the tests run faster.
-				[10];
```

Comment:
Let's use this pattern in the new tests too. As the comment says, when running perf tests specifically, we want the different sizes, but when the tests run as regular unit tests, having a single smaller size helps so unit test runtime doesn't explode.

Reply:
Applied